rust_src/src/frames.rs

```diff
@@ -346,4 +351,56 @@ pub fn frame_text_height(frame: LispObject) -> LispObject {
     LispObject::from(frame_or_selected(frame).text_height)
 }
 
+/// Return scroll bar width of FRAME in pixels.
+#[lisp_fn(min = "0")]
+pub fn frame_scroll_bar_width(frame: LispObject) -> LispObject {
+    let frame_ref = frame_or_selected(frame);
+    let width = match frame_ref.vertical_scroll_bar_type() {
+        vertical_scroll_bar_type::vertical_scroll_bar_left
+        | vertical_scroll_bar_type::vertical_scroll_bar_right => frame_ref.config_scroll_bar_width,
+        _ => 0,
+    };
```

Comment:
Why are you manually casting in each of these functions? Why not just return `EmacsInt`?

Reply:
```401 | pub fn frame_bottom_divider_width(frame: LispObject) -> EmacsInt {
    |                                                         -------- expected `i64` because of return type
402 |     let frame_ref = frame_or_selected(frame);
403 |     frame_ref.bottom_divider_width
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected i64, found i32
help: you can cast an `i32` to `i64`, which will sign-extend the source value
    |
403 |     frame_ref.bottom_divider_width.into()
```

`lisp_fn` should handle this right ?

Reply:
EmacsInt is a i64. The compiler warns about casts. So either `lisp_fn` is taught about i32 or you manually cast to an EmacsInt.

Reply:
If you want `lisp_fn` to handle it for you, your function must return a specific integer type like i32: `pub fn frame_bottom_divider_width(frame: LispObject) -> i32`

Reply:
IMO `lisp_fn` should be able to handle this.

Reply:
I see you already told me here to return i32. I think that was too early in the morning for me :P